File: IOS/Sources/OpenRoaming/OpenRoaming/Services/RefreshTokenService.swift
//
//  RefreshTokenService.swift
//  OpenRoaming
//
//  Created by Luis Vasquez on 05/06/20.
//  Copyright © 2020 Cisco. All rights reserved.
//

import Foundation

class RefreshTokenService: RegisterService {
    
    private var registerSdkHandler: RegisterSdkHandler!
    
    /**
    * Not implemented.
    * Use start(context: Context, acceptPrivacySettings: Boolean, serviceHandler: ServiceHandler).
    */
    override func start(serviceHandler: @escaping ServiceHandler) {
        fatalError("Not implemented")
    }
    
    override func start(handler: @escaping RegisterSdkHandler) {
        self.registerSdkHandler = handler
        onStart(headerParams: buildHeaderParams(sdkToken: SharedPrefs().sdkToken!), serviceHandler: {_,_,_ in })
   }

   override func onStart(headerParams: Array<HeaderParam>, serviceHandler: @escaping ServiceHandler) {
       let url = Urls.refreshTokenUrl()
       let requestData = RequestData(url: url, headerParams: headerParams, method: HTTPMethod.GET, body: nil)
       call(requestData: requestData, serviceHandler: { data, response, error in
            self.readSdkToken(data, response, error, self.registerSdkHandler)
       })
   }
}

File: Android/src/main/java/com/cisco/or/sdk/services/RefreshTokenService.kt
package com.cisco.or.sdk.services

import android.content.Context
import com.cisco.or.sdk.types.ServiceHandler
import com.cisco.or.sdk.utils.CryptoUtilAES
import com.cisco.or.sdk.utils.SharedPrefs
import com.cisco.or.sdk.utils.Urls

/**
 * Refresh the sdk token on the API.
 */
internal object RefreshTokenService : Service(){

    override fun start(context: Context, serviceHandler: ServiceHandler) {
        onStart(context, buildHeaderParams(context), serviceHandler)
    }

    override fun onStart(context: Context, headerParams: ArrayList<HeaderParam>, serviceHandler: ServiceHandler) {
        val url = Urls.refreshTokenUrl()

        val requestData = RequestData(url, headerParams, HTTP_METHOD.GET, null)
        call(context, requestData, RESPONSE_FORMATTER.JSON, serviceHandler)
    }
}